src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/service/CentralServerSystemParameterService.java

```diff
@@ -0,0 +1,126 @@
+    public static final String DEFAULT_CONF_SIGN_CERT_HASH_ALGO_URI = DigestMethod.SHA512;
+    public static final String SECURITY_SERVER_OWNERS_GROUP = "securityServerOwnersGroup";
+    public static final String DEFAULT_SECURITY_SERVER_OWNERS_GROUP = "security-server-owners";
+    public static final String DEFAULT_SECURITY_SERVER_OWNERS_GROUP_DESC = "Security server owners";
+    public static final String CONF_SIGN_DIGEST_ALGO_ID = "confSignDigestAlgoId";
+    public static final String DEFAULT_CONF_SIGN_DIGEST_ALGO_ID = CryptoUtils.SHA512_ID;
+    public static final String OCSP_FRESHNESS_SECONDS = "ocspFreshnessSeconds";
+    public static final Integer DEFAULT_OCSP_FRESHNESS_SECONDS = 3600;
+    public static final String TIME_STAMPING_INTERVAL_SECONDS = "timeStampingIntervalSeconds";
+    public static final Integer DEFAULT_TIME_STAMPING_INTERVAL_SECONDS = 60;
+    public static final String CONF_EXPIRE_INTERVAL_SECONDS = "confExpireIntervalSeconds";
+    public static final Integer DEFAULT_CONF_EXPIRE_INTERVAL_SECONDS = 600;
+    private static final String[] NODE_LOCAL_PARAMETERS = {CENTRAL_SERVER_ADDRESS};
+
+    private final SystemParameterRepository systemParameterRepository;
+    private final HAConfigStatus currentHaConfigStatus;
+
+    public String getParameterValue(String key, String defaultValue) {
+        List<SystemParameter> valueInDb;
+        if (currentHaConfigStatus.isHaConfigured()
+                && isNodeLocalParameter(key)
+        ) {
+            valueInDb = systemParameterRepository.findSystemParameterByKeyAndHaNodeName(
+                    key,
+                    currentHaConfigStatus.getCurrentHaNodeName()
+            );
+        } else {
+            valueInDb = systemParameterRepository.findSystemParametersByKey(key);
+        }
+        if (!valueInDb.isEmpty()) {
+            return valueInDb.iterator().next().getValue();
+        }
+        return defaultValue;
+    }
+
+    public SystemParameter updateOrCreateParameter(String lookupKey, String updateValue) {
+        Optional<SystemParameter> systemParameter;
+        if (currentHaConfigStatus.isHaConfigured() && isNodeLocalParameter(lookupKey)) {
+            String haNodeName = currentHaConfigStatus.getCurrentHaNodeName();
+            systemParameter = systemParameterRepository.findSystemParameterByKeyAndHaNodeName(lookupKey, haNodeName)
+                    .stream().findFirst();
+
+        } else {
+            systemParameter = systemParameterRepository.findSystemParametersByKey(lookupKey)
+                    .stream().findFirst();
+        }
+        if (systemParameter.isEmpty()) {
+            SystemParameter newSystemParameter = new SystemParameter();
+            newSystemParameter.setKey(lookupKey);
+            // now initial value for non-postgresql testing,
+            // the real haNodeName will be inserted using Postgresql database trigger.
+            newSystemParameter.setHaNodeName("node_0");
```

Comment:
Should probably use `currentHaConfigStatus.getCurrentHaNodeName()` and not hardcode `node_0`.

Reply:
Fixed in: 7e62b4a4d4d8df4ea9c90a62e4f9f21b3310b2f2